### ISSUE TITLE
MD076: List items with code blocks and tables with blank lines can't be used with 'tight' list setting

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#455]
### rumdl version

0.1.26

### Installation method

mise

### Bug description

`rumdl` can check list item spacing for consistency with rule MD076 (https://github.com/rvben/rumdl/blob/2d34df06e8361b822708099a38e3f5940fd81e85/docs/md076.md).

However, this rule might be used in conjunction with MD031 (https://github.com/rvben/rumdl/blob/2d34df06e8361b822708099a38e3f5940fd81e85/docs/md031.md) and MD058 (https://github.com/rvben/rumdl/blob/2d34df06e8361b822708099a38e3f5940fd81e85/docs/md058.md) for readability.

At the moment, if someone wants to use the default settings of MD031 and MD058, they can't set MD076 to enforce tight lists because the spacing around tables or code blocks looks like a "loose" list is wanted.

### Steps to reproduce

1. Create a Markdown file with this content:
   ````markdown
   # Heading

   Some introductory text.

   1. This is a list item with a fenced code block:

      ```markdown
      This is some Markdown
      ```

   1. This is a simple list item.
   1. This is a list item with a table:

      | Col 1 | Col 2 |
      |:------|:------|
      | Row 1 | Row 1 |
      | Row 2 | Row 2 |
   ````
1. Run `rumdl` with no specific configuration - use default settings.

### Expected behavior

I would expect to the Markdown above to not flag any violations. I want a "tight" list, but want spaces around code blocks and tables to maintain readability (and use the default settings of MD031 and MD058).

### Actual behavior

`rumdl` flags a violation at the third list item because it sits against the second list item. This Markdown passes:

````markdown
# Heading

Some introductory text.

1. This is a list item with a fenced code block:

   ```markdown
   This is some Markdown
   ```

1. This is a simple list item.

1. This is a list item with a table:

   | Col 1 | Col 2 |
   |:------|:------|
   | Row 1 | Row 1 |
   | Row 2 | Row 2 |
````

### Configuration

None - use default settings.

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.27 @eread. Blank lines around code blocks, tables, and HTML blocks inside list items are now treated as structural (required by MD031/MD058) rather than inter-item separators.
